lang/rust/avro/src/schema_compatibility.rs

```diff
@@ -96,10 +109,14 @@ impl Checker {
                         symbols: r_symbols, ..
                     }) = readers_schema
                     {
-                        return !w_symbols.iter().any(|e| !r_symbols.contains(e));
+                        if !w_symbols.iter().any(|e| !r_symbols.contains(e)) {
```

Comment:
Double negation give me headache ;-)
If i understand well, we must ensure that all element of w_symbols must be known from r_symbols ?
Is this
`if w_symbols.iter().all(|e| r_symbols.contains(e)) {`
would be same or not ??

Reply:
Yes, all the symbols in the writer should be in the reader, I think you suggestion makes sense and it is way better.